src/components/Profile/components/SchedulePanel/index.tsx

```diff
@@ -48,11 +49,27 @@ const GordonSchedulePanel = ({ profile, myProf }: Props) => {
     });
   }, [profile.AD_Username]);
 
+  useEffect(() => {
+    const getAccordionState = localStorage.getItem('accordionState');
+    if (getAccordionState !== null) {
+      setAccordionState(getAccordionState === 'true');
+    }
+  }, []);
+
+  const toggleAccordionState = () => {
+    setAccordionState(!accordionState);
+    localStorage.setItem('accordionState', String(!accordionState));
```

Comment:
This is actually introducing a subtle bug, due to the way React queues state updates. This is one of the major footguns of React, unfortunately. It's unlikely to matter in this case, but it's worth doing correctly to prevent user confusion. Also, it's good to learn about it now so you can avoid introducing more impactful bugs in the future.

Rather than spend several minutes writing a confusing explanation of the bug, I will point you to the React docs where they explain this common issue and how to work around it: [https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state)

The extremely short version is that passing a value to `setState` that is computed from the current state (e.g. the negation of the current `accordionState` in this case) will not always work the way you think it should. The correct way is to instead pass an _updater function_ that receives the current state as a parameter and uses that parameter to compute the new state, like so:
```suggestion
    setAccordionState((wasOpen) => !wasOpen);
    localStorage.setItem('accordionState', String(!accordionState));
```

Moreover, to ensure that localStorage stays in-sync with React's state, you should call `setItem` within the updater function:
```suggestion
    setAccordionState((wasOpen) => {
        localStorage.setItem('accordionState', String(!wasOpen));
        return !wasOpen;
    });
```